front/lib/api/vaults.ts

```diff
@@ -34,16 +34,18 @@ export const getDataSourceInfos = async (
 ): Promise<DataSourceOrViewInfo[]> => {
   const dataSources = await DataSourceResource.listByVault(auth, vault);
 
-  return dataSources.map((dataSource) => getDataSourceInfo(dataSource));
+  return Promise.all(
+    dataSources.map((dataSource) => getDataSourceInfo(auth, dataSource))
```

Comment:
How efficient is the usage query?

Reply:
can be improved : 

```
explain SELECT "agent_data_source_configuration"."id" AS "count" FROM "agent_data_source_configurations" AS "agent_data_source_configuration" INNER JOIN "data_source_views" AS "dataSourceView" ON "agent_data_source_configuration"."dataSourceViewId" = "dataSourceView"."id" AND "dataSourceView"."workspaceId" = 1 INNER JOIN "agent_retrieval_configurations" AS "agent_retrieval_configuration" ON "agent_data_source_configuration"."retrievalConfigurationId" = "agent_retrieval_configuration"."id" INNER JOIN "agent_configurations" AS "agent_retrieval_configuration->agent_configuration" ON "agent_retrieval_configuration"."agentConfigurationId" = "agent_retrieval_configuration->agent_configuration"."id" AND "agent_retrieval_configuration->agent_configuration"."status" = 'active' WHERE "agent_data_source_configuration"."dataSourceViewId" = 10;
                                                                                           QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Nested Loop  (cost=6.61..20.43 rows=1 width=4)
   ->  Nested Loop  (cost=6.61..19.35 rows=1 width=8)
         ->  Hash Join  (cost=6.46..16.73 rows=1 width=4)
               Hash Cond: (agent_retrieval_configuration."agentConfigurationId" = "agent_retrieval_configuration->agent_configuration".id)
               ->  Seq Scan on agent_retrieval_configurations agent_retrieval_configuration  (cost=0.00..10.20 rows=20 width=8)
               ->  Hash  (cost=6.44..6.44 rows=2 width=4)
                     ->  Seq Scan on agent_configurations "agent_retrieval_configuration->agent_configuration"  (cost=0.00..6.44 rows=2 width=4)
                           Filter: ((status)::text = 'active'::text)
         ->  Index Scan using agent_data_source_configurations_retrieval_configuration_id on agent_data_source_configurations agent_data_source_configuration  (cost=0.15..2.61 rows=1 width=12)
               Index Cond: ("retrievalConfigurationId" = agent_retrieval_configuration.id)
               Filter: ("dataSourceViewId" = 10)
   ->  Seq Scan on data_source_views "dataSourceView"  (cost=0.00..1.07 rows=1 width=4)
         Filter: ((id = 10) AND ("workspaceId" = 1))
(13 rows)
```

I also notice that the query we have is not checking the process_actions , it should be updated - we may have to do 2 separate queries .. ?

Reply:
Indeed. Playing a bit more with it, I feel like those screens are pretty slow to load already. Should we consider caching?

Reply:
created a task here : https://github.com/dust-tt/tasks/issues/1152